imagekit-sdk/src/test/java/io/imagekit/sdk/ImageKitTest.java

```diff
@@ -2,629 +2,1890 @@
+				+ "    \"width\": 236,\n" + "    \"size\": 23023,\n" + "    \"hasAlpha\": false,\n"
+				+ "    \"mime\": \"image/jpeg\"\n" + "}"));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		FileUpdateRequest fileUpdateRequest = new FileUpdateRequest("62a9c3ccd875ec6fd658c854");
+		List<String> tags = new ArrayList<>();
+		tags.add("Software");
+		tags.add("Developer");
+		tags.add("Engineer");
+		fileUpdateRequest.setTags(tags);
+
+		SUT.updateFileDetail(fileUpdateRequest);
+		RecordedRequest request = server.takeRequest();
+
+		String requestJson = "{\"fileId\":\"62a9c3ccd875ec6fd658c854\",\"tags\":[\"Software\",\"Developer\",\"Engineer\"]}";
+		String utf8RequestBody = request.getBody().readUtf8();
+		assertEquals(requestJson, utf8RequestBody);
+		assertEquals("application/json; charset=utf-8", request.getHeader("Content-Type"));
+		assertEquals("PATCH /v1/files/62a9c3ccd875ec6fd658c854/details HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.API_BASE_URL.concat("v1/files/62a9c3ccd875ec6fd658c854/details"),
+				request.getRequestUrl().toString());
+	}
+
+	@Test
+	public void imageKit_getFileList_returnList()
+			throws InterruptedException, ForbiddenException, TooManyRequestsException, InternalServerException,
+			UnauthorizedException, BadRequestException, UnknownException, IOException {
+
+		MockWebServer server = new MockWebServer();
+		String responseJson = "[\n" + "    {\n" + "        \"type\": \"file\",\n"
+				+ "        \"name\": \"default-image.jpg\",\n"
+				+ "        \"createdAt\": \"2022-06-11T07:26:19.294Z\",\n"
+				+ "        \"updatedAt\": \"2022-06-11T07:26:19.600Z\",\n"
+				+ "        \"fileId\": \"62a4439bce686814dfcce65c\",\n" + "        \"tags\": null,\n"
+				+ "        \"AITags\": null,\n" + "        \"versionInfo\": {\n"
+				+ "            \"id\": \"62a4439bce686814dfcce65c\",\n" + "            \"name\": \"Version 1\"\n"
+				+ "        },\n" + "        \"embeddedMetadata\": {\n"
+				+ "            \"DateCreated\": \"2022-06-11T07:26:19.599Z\",\n"
+				+ "            \"DateTimeCreated\": \"2022-06-11T07:26:19.600Z\"\n" + "        },\n"
+				+ "        \"customCoordinates\": null,\n" + "        \"customMetadata\": {},\n"
+				+ "        \"isPrivateFile\": false,\n"
+				+ "        \"url\": \"https://ik.imagekit.io/zv3rkhsym/default-image.jpg\",\n"
+				+ "        \"thumbnail\": \"https://ik.imagekit.io/zv3rkhsym/tr:n-ik_ml_thumbnail/default-image.jpg\",\n"
+				+ "        \"fileType\": \"image\",\n" + "        \"filePath\": \"/default-image.jpg\",\n"
+				+ "        \"height\": 1000,\n" + "        \"width\": 1000,\n" + "        \"size\": 147022,\n"
+				+ "        \"hasAlpha\": false,\n" + "        \"mime\": \"image/jpeg\"\n" + "    }\n" + "]";
+		server.enqueue(new MockResponse().setBody(responseJson));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		Map<String, String> options = new HashMap<>();
+		options.put("limit", "" + 1);
```

Comment:
Other parameters are not included. Also if the user passes `tags` as an array, will it be converted into a comma-separated string by SDK. All this needs to be asserted in test cases.